pkg/machine/shim/host.go

```diff
@@ -447,6 +450,30 @@ func Start(mc *vmconfigs.MachineConfig, mp vmconfigs.VMProvider, dirs *machineDe
 		return fmt.Errorf("reload config: %w", err)
 	}
 
+	// if the machine cannot continue starting due to a signal, ensure the state
+	// reflects the machine is no longer starting
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt, syscall.SIGTERM)
+	ctx, cancel := context.WithCancel(context.Background())
+	go func(ctx context.Context) {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				sig := <-signalChan
```

Comment:
After re-reading this  I'm not a fan of the implementation. looking at different ways to re-write it